uber/tests/email/email_fixtures.py

```diff
@@ -0,0 +1,165 @@
+    IDENT_TO_FIND = 'you_are_not_him'
+
+E = EmailTestsConstants
+
+
+@pytest.fixture
+def remove_all_email_categories(monkeypatch):
+    monkeypatch.setattr(AutomatedEmail, 'instances', OrderedDict())
+
+
+@pytest.fixture
+def add_test_email_categories(remove_all_email_categories):
+    AutomatedEmail(
+        model=Attendee,
+        subject='{EVENT_NAME}: You Need To Know',
+        ident='you_are_not_him',
+        template='unrest_in_the_house_of_light.html',
+        filter=lambda a: a.paid == c.NEED_NOT_PAY,
+        when=(),
+        sender="thomas.light@200X.com",
+        extra_data=None,
+        cc="proto@man.com",
+        bcc=None,
+        post_con=False,
+        needs_approval=True,
+    )
+
+
+@pytest.fixture
+def amazon_send_email_mock():
+    """
+    Patch the actual low-level method that actually sends an email out of our system onto the internet.
+    If this is called, you know that an email was really sent by our email subsystem.
+    """
+    with patch.object(AmazonSES, 'sendEmail', return_value=None) as mock:
+        yield mock
+
+
+@pytest.fixture
+def log_unsent_because_unapproved():
+    with patch.object(SendAllAutomatedEmailsJob, 'log_unsent_because_unapproved', return_value=None) as mock:
+        yield mock
+
+
+@pytest.fixture
+def setup_fake_test_attendees(monkeypatch):
+    # replace all email categories in the system with an empty list so we can add to it later
+    monkeypatch.setattr(AutomatedEmail, 'queries', {
+        Attendee: lambda ignored_param: [
+            Attendee(
+                placeholder=True,
+                first_name="Gambler",
+                last_name="Kirkdouglas",
+                email="thegambler@protos.com",
+                paid=c.NEED_NOT_PAY,
+                badge_type=c.GUEST_BADGE,
+                id=78,
+            ),
+            Attendee(
+                placeholder=True,
+                first_name="Kilroy",
+                last_name="Kilroy",
+                email="that_one_robot@ihaveasecret.com",
+                paid=c.NEED_NOT_PAY,
+                badge_type=c.GUEST_BADGE,
+                id=78,
+            ),
+            Attendee(
+                placeholder=False,
+                first_name="Reanimator",
+                last_name="Lovejoy",
+                email="yeswecan@jumpfromanywhere.com",
+                paid=c.HAS_PAID,
+                badge_type=c.ATTENDEE_BADGE,
+                id=12,
+            ),
+        ],
+        # Group: lambda ignored_param: would need to replace with: session.query(Group).options(subqueryload(Group.attendees))
+    })
+
+
+@pytest.fixture
+def set_now_to_sept_15th(monkeypatch):
+    return patch('uber.utils.localized_now', return_value=datetime(year=2016, month=9, day=15, hour=12, minute=30))
+
+
+@pytest.fixture
+def email_subsystem_sane_config(monkeypatch):
+    monkeypatch.setattr(c, 'DEV_BOX', False)
+    monkeypatch.setattr(c, 'SEND_EMAILS', True)
```

Comment:
I recommend injecting ``amazon_send_email_mock`` into this fixture, because that ensures that any unit test which uses this fixture will also have actual sending turned off, which makes extra sure that we don't make a mistake and send emails during a unit test.